adam/adam_6250.py

```diff
@@ -0,0 +1,352 @@
+
+    def read_PulseOutputHigh(self):
+        return self.pulse_output_high
+
+    def write_PulseOutputHigh(self, value):
+        self.connected_ADAM.write_registers(30, self.encode_values(value))
+
+    def read_AbsolutePulse(self):
+        return self.absolute_pulse_output
+
+    def write_AbsolutePulse(self, value):
+        self.connected_ADAM.write_registers(44, self.encode_values(value))
+
+    def read_IncrementalPulse(self):
+        return self.incremental_pulse_output
+
+    def write_IncrementalPulse(self, value):
+        self.connected_ADAM.write_registers(58, self.encode_values(value))
+
+    # ----------
+    # Attributes
+    # ----------
+
+    DigitalInput = attribute(
+        dtype=(bool,),
+        access=AttrWriteType.READ,
+        doc="Bool values at all Digital Input channels",
+        max_dim_x=8,
+    )
+
+    DigitalOutput = attribute(
+        dtype=(bool,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Bool values at all Digital Output channels",
+        max_dim_x=7
+    )
+
+    Counter = attribute(
+        dtype=(bool,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Counter status at all Digital Input channels",
+        max_dim_x=8
+    )
+
+    Overflow = attribute(
+        dtype=(bool,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Overflow flags at all Digital Input channels",
+        max_dim_x=8
+    )
+
+    LatchStatus = attribute(
+        dtype=(bool,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Latch Status at all Digital Input channels",
+        max_dim_x=8
+    )
+
+    CounterFrequency = attribute(
+        dtype=(int,),
+        access=AttrWriteType.READ,
+        doc="Counter / frequency values at all Digital Input channels",
+        max_dim_x=8
+    )
+
+    PulseOutputLow = attribute(
+        dtype=(int,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Pulse Output Low Level at all Digital Output channels",
+        max_dim_x=7
+    )
+
+    PulseOutputHigh = attribute(
+        dtype=(int,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Pulse Output High Level at all Digital Output channels",
+        max_dim_x=7
+    )
+
+    AbsolutePulse = attribute(
+        dtype=(int,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Absolute Pulse at all Digital Output channels",
+        max_dim_x=7
+    )
+
+    IncrementalPulse = attribute(
+        dtype=(int,),
+        access=AttrWriteType.READ_WRITE,
+        doc="Incremental Pulse at all Digital Output channels",
+        max_dim_x=7
+    )
+
+
+
+
+    # ---------------
+    # General methods
+    # ---------------
+
+    def init_device(self):
+        """Initiate device and sets its state to STANDBY"""
+        Device.init_device(self)
+        self.set_state(DevState.STANDBY)
+        self.set_status(
+            "ADAM-6250 in state STANDBY, ready to connect to "
+            "device")
+
+    def delete_device(self):
+        """Delete device"""
```

Comment:
Well, if you delete the device, you don't need to set its state and status, because it is going to be deleted when this method finishes!
I was thinking only of updating the docstring.